db.go

```diff
@@ -97,79 +97,40 @@ func addUserInfo(email string, userInfo []byte, d *gorm.DB) error {
 	}
 	return nil
 }
-
 /**
-Save Certificate Details key domain, value certDetails
+Save Certificate Details
 */
-func addCertDetails(domain string, certDetails []byte, d *gorm.DB) error {
-	s := string(certDetails)
-	if dbErr := d.Create(&CertDetail{Domain: domain, Value: s}).Error; dbErr != nil {
-		return errors.Wrapf(dbErr, "Unable to add cert details for %s", domain)
+func saveCertDetails(domain string, certDetailsRaw []byte, db *gorm.DB) error {
+	var certDetails CertDetail
+
+	if err := db.Where(&CertDetail{Domain: domain}).FirstOrInit(&certDetails).Error; err != nil {
+		return errors.Wrapf(err, "Unable to initialize cert details for %s", domain)
```

Comment:
At first read, I don't think this does what you're looking for.  At the end of the day, we need an upsert, right?  Seems that the Assign/FirstOrCreate pair is what you want:

```db.Where(User{Name: "jinzhu"}).Assign(User{Age: 30}).FirstOrCreate(&user)```

Look at http://gorm.io/docs/query.html#Assign-1

Reply:
Yeah, exactly, need the `upsert`. I like that :)

My testing on this worked, but I had to add the extra Save(). I'll take a look at this.

Reply:
Oh! I missed the trailing Save, but it looks like the above is gorm'ish way to do it.